src/app/auth/auth-interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { KeycloakService } from 'keycloak-angular';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private keycloak: KeycloakService) {}
+
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    // Get the access token from KeycloakService
+    const accessToken = this.keycloak.getKeycloakInstance().token;
```

Comment:
you should get it once then store in the localstorage, then get from there...if empty there then get from keycloak service instead